src/democracys-library.ts

```diff
@@ -1,33 +1,333 @@
+import { carousel1, carousel1Title } from './data/carousel-1';
+import { carousel2, carousel2Title } from './data/carousel-2';
+import type { CarouselCard } from './data/carousel-1';
+import { headerImageUrl } from './data/header-image';
+import { didYouKnow, Factoid } from './data/did-you-know';
+import './ti-tle';
 
 @customElement('ia-democracys-library')
 export class IaDemocracysLibrary extends LitElement {
-  @property({ type: String }) title = 'Hey there';
+  @property({ type: Array }) carousel1: CarouselCard[] = carousel1;
 
-  @property({ type: Number }) counter = 5;
+  @property({ type: Array }) carousel2: CarouselCard[] = carousel2;
 
-  private increment() {
-    this.counter += 1;
+  @property({ type: Array }) didYouKnow: Factoid[] = didYouKnow;
+
+  @property({ type: Array }) highlights: card[] = collectionsToHighlight;
+
+  @property({ type: String }) fontSource = '';
+
+  firstUpdated(): void {
+    const fontStyle = document.createElement('style');
+    fontStyle.innerHTML = `
+    @font-face {
+      font-family: "Teko";
+      src: url(https://archive.org/cors/democracys-library/web-component/Teko-SemiBold.ttf);
+    }`;
+    fontStyle.setAttribute('id', 'teko-font');
+    document.head.appendChild(fontStyle);
+  }
+
+  disconnectedCallback(): void {
+    const fontLink = document.head.querySelector('#teko-font');
+    if (fontLink) {
+      document.head.removeChild(fontLink);
+    }
+  }
+
+  // <i><img src="https://archive.org/download/democracys-library/web-component/help.svg" alt="help icon"></i>
+  get factoids(): TemplateResult[] {
+    return this.didYouKnow.map((fact: Factoid, i) => {
+      const tintColor = i % 2 === 0 ? 'green' : 'yellow';
+      return html`
+        <article>
+          <div class="title">
+            <ti-tle class=${tintColor}><span>DID YOU KNOW?</span></ti-tle>
+          </div>
+          <p class="full-width">${fact.details}</p>
+        </article>
+      `;
+    });
+  }
+
+  resourceCard(
+    card: card,
+    tintColor: 'yellow' | 'green' | undefined
+  ): TemplateResult {
+    return html`
+      <article>
+        <div class="title">
+          <ti-tle class=${tintColor ?? ''}>${card.title}</ti-tle>
+        </div>
+        <item-preview-image
+          src=${card.image}
+          alt="preview image."
+          class=${tintColor ?? ''}
+        ></item-preview-image>
+        <p>${card.blurb}</p>
+        <a href=${card.link} tab="_blank"
+          >Browse the ${card.title} collection</a
+        >
+      </article>
+    `;
+  }
+
+  carouselCard(
+    card: CarouselCard,
+    tintColor: 'yellow' | 'green' | undefined
+  ): TemplateResult {
+    const url = `https://archive.org/details/${card.id}`;
+    return html`
+      <a href=${url}>
+        <item-preview-image
+          src=${card.image}
+          alt=${`Go to: ${card.title}`}
+          class=${tintColor ?? ''}
+        ></item-preview-image>
+      </a>
+    `;
+  }
+
+  get topCarousel(): TemplateResult {
```

Comment:
suggestion: maybe this can be separate component something like: 
```
<carousel 
  .title=${title} 
  .carouselArray=${this.carousel1} 
  .position=${top|bottom}>
</carousel>
```

Reply:
hi, re:
- `title`
  - i thought about it adding `title` as prop. but it doesn't really need the data marshalling of title name to both title & carousel elements since they can both be composed as siblings. + i am keeping "card" styling at top level, carousel element is really constrained to the styling of the carousel itself.
- `position`
  - i do not see this prop as any value?  the placement of carousel in web component is at the top composition, not inside of component...
- `carouselArray`
  - it's not really concise, the name refers to data's structure, not necessarily its responsibility. perhaps `items` seems more appropriate?
  - also, this is also feels like marshalling of data since the card template can be drawn at the same level as this component.
 
To have _another_ shadow nest seems a bit too strong for a display component.

i have to do a few more tweaks to the carousel and i will consider making an official component for the carousel behaviour only - excluding title and overall "card" design as it will fit into the card.

thanks for walking thru that!